File: TransitOps_frontend/src/types/vehicle.ts
import type { VehicleStatus } from './dashboard'

export type Vehicle = {
  id: number
  registrationNumber: string
  name: string
  type: string
  maxLoadCapacity: number
  odometer: number
  acquisitionCost: number
  status: VehicleStatus
}

export type VehicleRequest = {
  registrationNumber: string
  name: string
  type: string
  maxLoadCapacity: number
  odometer: number
  acquisitionCost: number
  status?: VehicleStatus
}

File: TransitOps_backend/src/main/java/com/transitops/entity/Vehicle.java
package com.transitops.entity;

import java.math.BigDecimal;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "vehicles")
@Getter
@Setter
@NoArgsConstructor
public class Vehicle {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column(nullable = false, unique = true, length = 50)
	private String registrationNumber;

	@Column(nullable = false, length = 100)
	private String name;

	@Column(nullable = false, length = 50)
	private String type;

	@Column(nullable = false)
	private Integer maxLoadCapacity;

	@Column(nullable = false)
	private Integer odometer;

	@Column(nullable = false, precision = 12, scale = 2)
	private BigDecimal acquisitionCost;

	@Enumerated(EnumType.STRING)
	@Column(nullable = false, length = 20)
	private VehicleStatus status = VehicleStatus.AVAILABLE;
}
